src/service/chat.js

```diff
@@ -28,6 +28,51 @@ const createChannel = async (channelType, members, channelName) => {
   }
 };
 
+const connectUser = async () => {
+  const token = TokenStorage.get(ITokenEnum.token);
+  const id = await getUserId();
+  const user = {id};
+  await chatClient.disconnectUser();
+  await chatClient.connectUser(user, token);
+};
+
+const queryChannels = async (members) => {
+  const sort = [{last_message_at: -1}];
+  const filter = {type: 'messaging', members: {$eq: members}};
+  const result = await chatClient.queryChannels(filter, sort, {watch: true, state: true});
+  return result;
+};
+
+const followClient = async (members, data, text, textOwnMessage) => {
+  try {
+    await connectUser();
+    const channels = await queryChannels(members);
+    const message = {
+      user_id: data.user_id_follower,
+      text,
+      isSystem: true,
+      silent: true,
+      textOwnMessage,
+      userIdFollowed: data?.user_id_followed,
+      userIdFollower: data?.user_id_follower
+    };
+
+    const name = [data?.username_followed, data?.username_follower].join(',');
+
+    if (channels?.length <= 0) {
+      const newChannel = await createChannel('messaging', members, name);
+      return newChannel.sendMessage(message, {skip_push: true});
+    }
+    const messageClient = chatClient.channel('messaging', channels[0].id);
+    await messageClient.sendMessage(message, {skip_push: true});
+
+    return channels;
+  } catch (error) {
+    console.log('error follow client', error);
+    throw error;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `followClient` function, there's a console.log statement for logging errors. While this might be useful during development, it's generally not a good practice to leave console.log statements in production code as they can expose sensitive information. Instead, consider using a proper logging mechanism or remove the console.log statement.
```diff
  } catch (error) {
-71:     console.log('error follow client', error);
+71:     // Consider using a proper logging mechanism here
    throw error;
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->